pgmpy/estimators/MLE.py

```diff
@@ -42,24 +43,30 @@ class MaximumLikelihoodEstimator(ParameterEstimator):
     """
 
     def __init__(self, model, data, **kwargs):
-        if not isinstance(model, BayesianNetwork) and not isinstance(
-            model, JunctionTree
-        ):
+        if not isinstance(model, (BayesianNetwork, JunctionTree, DAG)):
             raise NotImplementedError(
-                "Maximum Likelihood Estimate is only implemented for BayesianNetwork and JunctionTree"
+                "Maximum Likelihood Estimate is only implemented for BayesianNetwork, JunctionTree, and DAG"
             )
 
-        elif set(model.nodes()) > set(data.columns):
-            if isinstance(model, BayesianNetwork):
+        if isinstance(model, (DAG, BayesianNetwork)):
+            if len(model.latents) > 0:
                 raise ValueError(
-                    f"Found latent variables: {model.latents}. Maximum Likelihood doesn't support latent variables, please use ExpectationMaximization"
+                    f"Found latent variables: {model.latents}. Maximum Likelihood doesn't support latent variables, please use ExpectationMaximization."
                 )
-            else:
+
+            elif set(model.nodes()) > set(data.columns):
                 raise ValueError(
                     "Nodes detected in the model that are not present in the dataset. "
-                    + "Refine the model so that all parameters can be estimated from the data."
+                    "Refine the model so that all parameters can be estimated from the data."
```

Comment:
I think we can make this error more informative by also printing the missing variables in the error message.